### ISSUE TITLE
关于微调

### ISSUE THREAD

**Issue by kkkkkkb** [AILab-CVC/YOLO-World#49]
您好，我在调试train.py的过程中，runner = Runner.from_cfg(cfg)会卡住无法向下运行，请问有什么解决办法吗？还有一个问题就是我在自己的数据集(10000张图，只有一类baby)上进行微调时，发现定位loss一直是0，几个epoch后分类loss也降为0，但是我训练了40个epoch后，用image_demo进行测试发现无法检测到目标(图中的目标一个都检测不到)，想问下您有没有一些建议。

**Comment by wondervictor** [comment]
您好，非常抱歉回复较晚。
针对第一个问题`runner = Runner.from_cfg(cfg)会卡住无法向下运行`,这个请问可以kill的时候知道卡在哪一步吗？
针对第二个问题: 这里可能是标注不一致的问题，可以先确定一下box的标注是不是和coco一致，coco默认用的是xywh(xy是左上角的点)
<img width="824" alt="image" src="https://github.com/AILab-CVC/YOLO-World/assets/15026693/321b0d86-fdd7-4455-91b3-f0c019b8848b">

**Comment by kkkkkkb** [comment]
@wondervictor 
您好，非常感谢您的回复！
我发现检测不到目标的原因应该是数据读取的问题，使用coco的json文件时是可以读取到bbox等标注信息的，但是使用自己的json文件时却读不到bbox等标注信息，但是两个json文件的区别好像只是segmentation，希望您有时间可以帮我看一下，十分感谢！
coco的json文件：
`"annotations": [{"segmentation": [[510.66,423.01,511.72,420.03,510.45,416.0,510.34,413.02,510.77,410.26,510.77,407.5,510.34,405.16,511.51,402.83,511.41,400.49,510.24,398.16,509.39,397.31,504.61,399.22,502.17,399.64,500.89,401.66,500.47,402.08,499.09,401.87,495.79,401.98,490.59,401.77,488.79,401.77,485.39,398.58,483.9,397.31,481.56,396.35,478.48,395.93,476.68,396.03,475.4,396.77,473.92,398.79,473.28,399.96,473.49,401.87,474.56,403.47,473.07,405.59,473.39,407.71,476.68,409.41,479.23,409.73,481.56,410.69,480.4,411.85,481.35,414.93,479.86,418.65,477.32,420.03,476.04,422.58,479.02,422.58,480.29,423.01,483.79,419.93,486.66,416.21,490.06,415.57,492.18,416.85,491.65,420.24,492.82,422.9,493.56,424.39,496.43,424.6,498.02,423.01,498.13,421.31,497.07,420.03,497.07,415.15,496.33,414.51,501.1,411.96,502.06,411.32,503.02,415.04,503.33,418.12,501.1,420.24,498.98,421.63,500.47,424.39,505.03,423.32,506.2,421.31,507.69,419.5,506.31,423.32,510.03,423.01,510.45,423.01]],"area": 702.1057499999998,"iscrowd": 0,"image_id": 289343,"bbox": [473.07,395.93,38.65,28.67],"category_id": 18,"id": 1768},`
我的json文件：
"annotations":[{"id":0,"image_id":0,"category_id":1,"bbox":[328,176,158.5,403],"area":63875.5,"segmentation":[],"iscrowd":0},
下面是我尝试分别读取coco数据集和自己数据集的图片信息，发现在读取自己数据集的时候"instance"为空：
coco数据集：
![image](https://github.com/AILab-CVC/YOLO-World/assets/124493154/096b465f-7c39-4f7e-ad68-af1206e4208f)
自己数据集：
![image](https://github.com/AILab-CVC/YOLO-World/assets/124493154/9305e5b6-6d2c-4978-bbb8-a1e00c27be8d)

**Comment by kkkkkkb** [comment]
补充一下，这里提到的两个配置文件我都进行了尝试，结果还是一样的：
![image](https://github.com/AILab-CVC/YOLO-World/assets/124493154/27c2cec2-9408-4222-a4a5-271c5cda5e04)

**Comment by wondervictor** [comment]
您好，你需要根据 https://github.com/AILab-CVC/YOLO-World/issues/52 去删除mask-refine相关的参数，可能因为没有segmentation会过滤掉box的标签，您这边没有segmentation标注，就直接用下面这个without mask annotations的吧。考虑到大家下游训练没有mask，我们后续提供一些 `w/o mask-refine`的config样例。

**Comment by kkkkkkb** [comment]
@wondervictor 
您好，非常感谢您的耐心回复！
我搜索了一下self.use_mask_refine，发现只有在两个transform文件中的YOLOv5RandomAffine类中有用到，但是都只是在类的初始化函数__init__()中用到了，后续并没有其他使用，而且我也根据#52问题将use_mask_refine修改为了False，问题还是没有解决。
![image](https://github.com/AILab-CVC/YOLO-World/assets/124493154/896b8858-65e5-4825-8295-ead9fa97af39)
`mosaic_affine_transform = [
    dict(type='MultiModalMosaic',
         img_scale=_base_.img_scale,
         pad_val=114.0,
         pre_transform=_base_.pre_transform),
    dict(type='YOLOv5CopyPaste', prob=_base_.copypaste_prob),
    dict(
        type='YOLOv5RandomAffine',
        max_rotate_degree=0.0,
        max_shear_degree=0.0,
        max_aspect_ratio=100.,
        scaling_ratio_range=(1 - _base_.affine_scale, 1 + _base_.affine_scale),
        # img_scale is (width, height)
        border=(-_base_.img_scale[0] // 2, -_base_.img_scale[1] // 2),
        border_val=(114, 114, 114),
        min_area_ratio=_base_.min_area_ratio,
        use_mask_refine=False)`
期待您的w/o mask-refine的config样例！

**Comment by wondervictor** [comment]
@kkkkkkb 您好，w/o mask-refine的config已经上传了，您可以参考一下 [yolo_world_l_dual_vlpan_2e-4_80e_8gpus_finetune_coco.py](https://github.com/AILab-CVC/YOLO-World/blob/master/configs/finetune_coco/yolo_world_l_dual_vlpan_2e-4_80e_8gpus_finetune_coco.py)。

**Comment by kkkkkkb** [comment]
@wondervictor 我试了下将两个dataset里的filter_cfg一行删掉了，因为是默认为None，但还是读不到标签，instances还是空
`coco_train_dataset = dict(
    _delete_=True,
    type='MultiModalDataset',
    dataset=dict(
        type='YOLOv5CocoDataset',
        data_root='multi-modal/YOLO-World-master/data/family',
        ann_file='annotations/instances_train2017.json',
        data_prefix=dict(img='train2017/')),
    class_text_path='multi-modal/YOLO-World-master/data/texts/family_class_texts.json',
    pipeline=train_pipeline)`

**Comment by kkkkkkb** [comment]
@wondervictor 十分感谢您的耐心解答，我终于找到问题了，确实是self.cat_ids那里的问题，一开始self.cat_ids一直为空，导致了读不到标签，thank you！！！

**Comment by wondervictor** [comment]
@kkkkkkb 不客气，如果您有新的问题，可以重新打开一个issues，感谢您对YOLO-World的关注！
